liblouis/lou_translateString.c

```diff
@@ -408,6 +409,10 @@ swapTest(int swapIC, int *pos, const TranslationTableHeader *table, const InStri
 	}
 	while (p - *pos < passInstructions[swapIC + 4]) {
 		int test;
+		if (p >= input->length) {
```

Comment:
Sorry for my ignorance, but why in the first case you return 0 when `p` is past the input length whereas in the second case you return 1 and even set `*pos` to `p` (which is past the input length). This all seems to be in the same function.

Reply:
I initially returned 0 indeed, but got bogus test results :)

It does make sense return 1 and set *pos to p, see the

```
if (test >= swapRule->charslen) {
	*pos = p;
	return 1;
}
```

part below, which is achieved when the match is not found (and in the previous loop it was just returning 0). Seeing the input terminate early is also a match-not-found, so we have to do the same.

Reply:
OK, makes sense

Reply:
Maybe it would help if we explain the `passInstructions` data structure somewhere. `passInstructions[swapIC + 3]` is probably the lower bound and passInstructions[swapIC + 4] to upper bound of the number of characters.